### PR TITLE
Fix the deadlock bug in catalog

### PR DESCRIPTION
**Pull request by zzbdatatom**
PR checklist:
- Did you run ClangFormat ? yes
- Did you run CheckStyle ? yes
- Did you check the comment / log / exception conventions in Engineering code process wiki page ? yes
- Did you import unnecessary headers ? no 
- Did you surround `Daisy : starts/ends` for new code in existing ClickHouse code base ? yes

Please write user-readable short description of the changes:
1. Fix the deadlock problem caused by deleting the storage data in catalogService
2. Fix the bug that some configurations of Kafka did not take effect

### CODE REVIEW

File: src/DistributedWriteAheadLog/KafkaWAL.cpp [hunk 1]
```diff
@@ -273,14 +273,26 @@ void KafkaWAL::backgroundPollProducer() const
...
+        /// Maximum number of messages allowed on the producer queue.
         std::make_pair("queue.buffering.max.messages", std::to_string(settings->queue_buffering_max_messages)),
+        std::make_pair("queue.buffering.max.kbytes", std::to_string(settings->queue_buffering_max_kbytes)),
         std::make_pair("queue.buffering.max.ms", std::to_string(settings->queue_buffering_max_ms)),
         std::make_pair("message.send.max.retries", std::to_string(settings->message_send_max_retries)),
+        /// The backoff time in milliseconds before retrying a protocol request.
         std::make_pair("retry.backoff.ms", std::to_string(settings->retry_backoff_ms)),
+        /// the producer will ensure that messages are successfully produced exactly once
         std::make_pair("enable.idempotence", std::to_string(settings->enable_idempotence)),
+        // librdkafka statistics emit interval time.
```
Comment (hzhzhang) [inline src/DistributedWriteAheadLog/KafkaWAL.cpp]: let's keep the comment in the same style.

Comment (zzbdatatom) [reply src/DistributedWriteAheadLog/KafkaWAL.cpp]: fixed

File: src/DistributedMetadata/CatalogService.cpp [hunk 2]
```diff
@@ -703,13 +703,18 @@ void CatalogService::mergeCatalog(const NodePtr & node, TableContainerPerNode sn
             iter_by_name->second.insert_or_assign(std::move(node_shard), p.second);
         }
 
+        /// FIXME, if table definition changed, we will need update the storage inline
         {
```
Comment (chenziliang) [inline src/DistributedMetadata/CatalogService.cpp:708]: we don't need the outer `{}`

File: src/DistributedWriteAheadLog/KafkaWALConsumer.cpp [hunk 3]
```diff
@@ -75,7 +75,7 @@ void KafkaWALConsumer::initHandle()
         std::make_pair("bootstrap.servers", settings->brokers.c_str()),
         std::make_pair("group.id", settings->group_id),
         /// Enable auto offset commit
-        std::make_pair("enable.auto.commit", "true"),
+        std::make_pair("enable.auto.commit", std::to_string(settings->enable_auto_commit)),
```
Comment (chenziliang) [inline src/DistributedWriteAheadLog/KafkaWALConsumer.cpp]: we shall never make this as `false`, we can remove this flag in settings

File: src/DistributedMetadata/CatalogService.cpp (hunk 2 above)
Comment (zzbdatatom) [reply src/DistributedMetadata/CatalogService.cpp:708]: fixed

File: src/DistributedWriteAheadLog/KafkaWALConsumer.cpp (hunk 3 above)
Comment (zzbdatatom) [reply src/DistributedWriteAheadLog/KafkaWALConsumer.cpp]: fixed

File: src/DistributedWriteAheadLog/KafkaWALSettings.h [hunk 4]
```diff
@@ -58,7 +58,6 @@ struct KafkaWALSettings
     int32_t session_timeout_ms = 10000;
     int32_t max_poll_interval_ms = 30000;
 
-    bool enable_auto_commit = true;
```
Comment (chenziliang) [inline src/DistributedWriteAheadLog/KafkaWALSettings.h]: We can delete this setting, but we can't delete line 97. I would say keep both of them for now and don't expose this setting in config.xml

Comment (zzbdatatom) [reply src/DistributedWriteAheadLog/KafkaWALSettings.h]: fixed

**Review by chenziliang — APPROVED** [review]

**Pull request merged**